src/codemagic/tools/_app_store_connect/arguments.py

```diff
@@ -352,6 +361,22 @@ class AppStoreConnectArgument(cli.Argument):
         description='Turn on logging for App Store Connect API HTTP requests',
         argparse_kwargs={'required': False, 'action': 'store_true'},
     )
+    UNAUTHORIZED_REQUEST_RETRIES = cli.ArgumentProperties(
+        key='unauthorized_request_retries',
+        flags=('--api-unauthorized-retries', '-r'),
+        type=Types.ApiUnauthorizedRetries,
+        description=(
+            'Specify how many times should the App Store Connect API request '
+            'be retried in case the called request fails due to an '
+            'authentication error (401 Unauthorized response from the server). '
+            'In case unauthorized response is returned, then the new request '
+            'is attempted with a new JSON Web Token until number of attempts '
+            'is exhausted.'
```

Comment:
```suggestion
            'Specify how many times the App Store Connect API request '
            'should be retried in case the called request fails due to an '
            'authentication error (401 Unauthorized response from the server). '
            'In case of the above authentication error, the request is retried using'
            'a new JSON Web Token as many times until the number of retries '
            'is exhausted.'
```